project/frontend/src/pages/HomePage.js

```diff
@@ -1,21 +1,27 @@
 import './HomePage.css';
 
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { useState } from 'react';
+
+import Modal from '../components/Modal.js';
+import PreferenceForm from '../components/PreferenceForm.js';
 
 function HomePage() {
-  return (
-    <div className='container u-full-width'>
+  const [isPreferenceFormOpen, setIsPreferenceFormOpen] = useState(false);
+
+  const togglePreferenceForm = () => {
+    setIsPreferenceFormOpen((prev) => !prev);
+  };
+
+  return [
```

Comment:
Can you return an array like this in the render?

Reply:
Yes, apparently as of React 16.x.x you can return arrays instead of having to return a root node. It gets added to the DOM like you'd expect with the elements in the array being siblings. Alternatives are wrapping in an extra `div` or 'React.Fragment' component, but I think the array is okay and cuts down on nesting.